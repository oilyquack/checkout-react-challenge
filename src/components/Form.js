import React from "react";

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      name: "",
      email: "",
      comment: ""
    };

    this.handleNameChange = this.handleNameChange.bind(this);
    this.handleEmailChange = this.handleEmailChange.bind(this);
    this.handleCommentChange = this.handleCommentChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleNameChange(event) {
    event.preventDefault();

    this.setState({
      name: event.target.value
    });
  }

  handleEmailChange(event) {
    event.preventDefault();

    this.setState({
      email: event.target.value
    });
  }

  handleCommentChange(event) {
    event.preventDefault();

    this.setState({
      comment: event.target.value
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const name = this.state.name;
    const email = this.state.email;
    const comment = this.state.comment;

    const newReview = {
      name,
      email,
      comment
    };

    this.props.receiver(newReview);

    this.setState({
      name: "",
      email: "",
      comment: ""
    });

    document.querySelector("#name-input").value = "";
    document.querySelector("#email-input").value = "";
    document.querySelector("#comment-input").value = "";
  }

  render() {
    return (
      <form className="app__form" onSubmit={this.handleSubmit}>
        <label>Name</label>
        <input
          id="name-input"
          placeholder="Name"
          type="text"
          onChange={this.handleNameChange}
          required
        />
        <label>Email</label>
        <input
          id="email-input"
          placeholder="Email"
          type="email"
          onChange={this.handleEmailChange}
          required
        />
        <label>Comment</label>
        <input
          id="comment-input"
          placeholder="Comment"
          type="text"
          maxLength="280"
          onChange={this.handleCommentChange}
          required
        />
        <button type="submit">Submit</button>
      </form>
    );
  }
}

export default App;
