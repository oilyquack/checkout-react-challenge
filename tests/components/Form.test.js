import React from "react";
import renderer from "react-test-renderer";
import Form from "../../src/components/Form";

it("matches the snapshot", () => {
  const tree = renderer.create(<Form />);
  expect(tree).toMatchSnapshot();
});
