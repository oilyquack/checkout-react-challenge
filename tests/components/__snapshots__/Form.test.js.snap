// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<form
  className="app__input__form"
  onSubmit={[Function]}
>
  <label>
    Name
  </label>
  <input
    id="name-input"
    onChange={[Function]}
    placeholder="Name"
    required={true}
    type="text"
  />
  <label>
    Email
  </label>
  <input
    id="email-input"
    onChange={[Function]}
    placeholder="Email"
    required={true}
    type="email"
  />
  <label>
    Rating
  </label>
  <div
    id="app__form__rating"
  >
    <span
      className="star"
      id="1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      ☆
    </span>
    <span
      className="star"
      id="2"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      ☆
    </span>
    <span
      className="star"
      id="3"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      ☆
    </span>
    <span
      className="star"
      id="4"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      ☆
    </span>
    <span
      className="star"
      id="5"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      ☆
    </span>
  </div>
  <label>
    Comment
  </label>
  <textarea
    id="comment-input"
    onChange={[Function]}
    placeholder="Comment"
    required={true}
    type="text"
  />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
